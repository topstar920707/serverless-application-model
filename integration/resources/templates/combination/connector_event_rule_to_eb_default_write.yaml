Resources:
  TriggerFunction:
    Type: AWS::Serverless::Function
    Properties:
      Runtime: nodejs16.x
      Handler: index.handler
      Timeout: 10  # in case eb has delay
      InlineCode: |
        const AWS = require('aws-sdk');

        exports.handler = async (event) => {
          const eb = new AWS.EventBridge();
          const response = await eb.putEvents({
            Entries: [{
              Source: process.env.EVENT_SOURCE,
              Detail: "{}",
              DetailType: "Test",
              EventBusName: process.env.EVENT_BUS_NAME,
            }]
          }).promise();

          const sqs = new AWS.SQS();
          const data = await sqs.receiveMessage({
            QueueUrl: process.env.QUEUE_URL,
            WaitTimeSeconds: 5,
          }).promise();

          if (data.Messages.length == 0) {
            throw 'No messages in the queue!';
          }
        };
      Environment:
        Variables:
          QUEUE_URL: !Ref VerificationQueue
          EVENT_SOURCE: !Sub '${AWS::StackName}-test-event'
          EVENT_BUS_NAME: !Ref CustomEventBus
      Policies:
        - EventBridgePutEventsPolicy:
            EventBusName: !Ref CustomEventBus
        - SQSPollerPolicy:
            QueueName: !GetAtt VerificationQueue.QueueName

  CustomEventBus:
    Type: AWS::Events::EventBus
    Properties:
      Name: !Sub "${AWS::StackName}-EventBus"

  EventRule:
    Type: AWS::Events::Rule
    Properties:
      Description: !Sub 'EventRule-${AWS::StackName}'
      EventBusName: !Ref CustomEventBus
      EventPattern:
        source:
          - !Sub '${AWS::StackName}-test-event'
      Targets:
        - Arn: !Sub arn:${AWS::Partition}:events:${AWS::Region}:${AWS::AccountId}:event-bus/default
          Id: "EBBus"
          RoleArn: !GetAtt EBRole.Arn

  EBRole:
    Type: AWS::IAM::Role
    Properties:
      AssumeRolePolicyDocument:
        Version: '2012-10-17'
        Statement:
          - Effect: Allow
            Principal:
              Service: events.amazonaws.com
            Action: sts:AssumeRole

  MyConnector:
    Type: AWS::Serverless::Connector
    Properties:
      Source:
        Id: EventRule
      Destination:
        Type: AWS::Events::EventBus
        Arn: !Sub arn:${AWS::Partition}:events:${AWS::Region}:${AWS::AccountId}:event-bus/default
      Permissions:
        - Write

  VerificationEventRule:
    Type: AWS::Events::Rule
    Properties:
      Description: !Sub 'EventRule-${AWS::StackName}-verification'
      EventPattern:
        source:
          - !Sub '${AWS::StackName}-test-event'
      Targets:
        - Arn: !GetAtt VerificationQueue.Arn
          Id: "SQSqueue"

  VerificationQueue:
    Type: AWS::SQS::Queue

  ConnectorNotBeingTested:
    Type: AWS::Serverless::Connector
    Properties:
      Source:
        Id: VerificationEventRule
      Destination:
        Id: VerificationQueue
      Permissions:
        - Write
