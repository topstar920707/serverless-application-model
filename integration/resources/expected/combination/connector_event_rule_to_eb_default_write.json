[
  {
    "LogicalResourceId": "ConnectorNotBeingTestedQueuePolicy",
    "ResourceType": "AWS::SQS::QueuePolicy"
  },
  {
    "LogicalResourceId": "CustomEventBus",
    "ResourceType": "AWS::Events::EventBus"
  },
  {
    "LogicalResourceId": "EBRole",
    "ResourceType": "AWS::IAM::Role"
  },
  {
    "LogicalResourceId": "EventRule",
    "ResourceType": "AWS::Events::Rule"
  },
  {
    "LogicalResourceId": "MyConnectorPolicy",
    "ResourceType": "AWS::IAM::ManagedPolicy"
  },
  {
    "LogicalResourceId": "TriggerFunction",
    "ResourceType": "AWS::Lambda::Function"
  },
  {
    "LogicalResourceId": "TriggerFunctionRole",
    "ResourceType": "AWS::IAM::Role"
  },
  {
    "LogicalResourceId": "VerificationEventRule",
    "ResourceType": "AWS::Events::Rule"
  },
  {
    "LogicalResourceId": "VerificationQueue",
    "ResourceType": "AWS::SQS::Queue"
  }
]